jwst/stpipe/tests/test_input.py

```diff
@@ -12,7 +12,7 @@
 from jwst.stpipe.tests.steps import StepWithModel, StepWithContainer
 
 
-def test_default_input_with_container(mk_tmp_dirs):
+def test_default_input_with_container(_jail):
```

Comment:
Is `_jail` even necessary here?  I don't think `StepWithContainer` or `StepWithModel` will save anything by default, and my understanding of `get_pkg_data_filename` is that it doesn't need `_jail` in order to allow the test to see the correct data.

I have the same question about many of the tests below where this change is made.

Reply:
I agree that if it does not actually write out files, `_jail` is not necessary. I don't know the Step stuff enough to decide. My goal was getting rid of `mk_tmp_dirs` when it is not necessary because calling `tempfile` not once, but three times is too much for me to bear.

Reply:
Case-by-case study of this might be time consuming and will drag this PR out a long time. Should we just use `_jail` for now so at least we can control the tests with `basetemp` *and* prevent them from create 3 random temporary directories per run, and address the case by case stuff as follow-up issue?